ports/unix/main.c

```diff
@@ -553,10 +561,8 @@ MP_NOINLINE int main_(int argc, char **argv) {
             p++;
         }
     }
-    mp_obj_list_init(MP_OBJ_TO_PTR(mp_sys_path), path_num);
-    mp_obj_t *path_items;
-    mp_obj_list_get(mp_sys_path, &path_num, &path_items);
-    path_items[0] = MP_OBJ_NEW_QSTR(MP_QSTR_);
+    mp_sys_path = mp_obj_new_list(0, NULL);
+    mp_obj_list_append(mp_sys_path, MP_OBJ_NEW_QSTR(MP_QSTR_));
```

Comment:
Why not leave the `path_items` array as it was?  Ie:
```c
mp_sys_path = mp_obj_new_list(path_num, NULL);
mp_obj_t *path_items;
mp_obj_list_get(mp_sys_path, &path_num, &path_items);
...
```

Reply:
This was to support having both assignable and non-assignable sys (which was later removed).

I've just rewritten this to do a single loop. It's simpler now, and fixes a small bug where before this PR it did:

```
$ env 'MICROPYPATH=:~/a:b' ./build-standard/micropython -c 'import sys; print(sys.path)'
['', '', '/home/jimmo/a', 'b']
```
now correctly does
```
$ env 'MICROPYPATH=:~/a:b' ./build-standard/micropython -c 'import sys; print(sys.path)'
['', '/home/jimmo/a', 'b']
```

Reply:
OK